### ISSUE TITLE
Do not use IP address to identify devices (prefer MAC as identity)

### ISSUE THREAD

**Issue by troy216** [danmed/TasmoBackupV1#89]
I use DHCP with my Tasmota devices, so their IP addresses will change over time.  It looks like TasmotaBackup uses the IP address to identify devices.  This results in two issues:  

1. After re-discovering devices, I will get duplicate records where the Name and MAC address are the same but IP address is different.
2. When Iisting the backup files, often multiple device names will appear in the list.

I suggest that only the MAC address be used to identify devices so that there is never any confusion.  Alternatively, either the device name or the MQTT topic could be used.  The IP address should NEVER be used to determine identity.

**Comment by troy216** [comment]
Sorry if I came across blunt.  I'm sure there are many design considerations I am not aware of.  Also, I am generally very pleased with this addon and appreciate your contribution.

In the readme, it says, "No duplicates (based on IP, and MAC)".  I'm not sure exactly what that means, but because I ended up with duplicates where only the IP was different (MAC and Name were the same), I assumed it was using the IP, in some way, to identify devices.  

In my case, I replaced my wifi router, so that is why all my devices received different IP addresses.  While many devices were correctly identified, a handful of duplicates were created. 

If only the MAC were used to identify devices, then duplicates should never occur, and backup files should be correctly associated with the device.  The only scenario where this would not work ideally is when replacing a device with an alternate device, say because it failed.  That is why the device Name (I am using MQTT topic to populate name) might be a preferred identifier.  However, I would vote to use the MAC address alone.  I can always manually apply the backup to the new hardware if replacing a failed device.

**Comment by patrickdk77** [comment]
Yes, it attempts to do it by mac if it can figure out the mac address, then it should go by ip.
It might not have the mac address for many reasons, tasmoto version did not support fetching that information, the device is on a different network than tasmobackup. It attempts to fill in mac when it does find it. It is a best effort though.

I'll have to review the code and see if there are cases it should work, but isn't.

**Comment by heffneil** [comment]
I have this complaint as well.  I repurposed some previously installed switches.  When I went to re-add them they got an old IP for a device that also went away.  When I went to back it up it said it existed.  When I removed the device with the IP - it then let me add it so it wasn't hung up on mac address but IP.  Ip isn't a reliable form of identifying devices.

**Comment by patrickdk77** [comment]
I just redid my network, and my iot network has whole new ip subnet, I had no issues, everything was mapped by mac correctly, as IP is only used if there is no mac address recorded in the database for the device.

I am not sure what else I can do to attempt to replicate this, but will see.
The mac address code has been in here awhile and maybe the ip code should be removed (except it will cause issues with older tasmota versions then), but if your having issues now from IP, if I removed IP nothing will work, since IP is only used if MAC does not exists.

If I go into my tasmobackup database (tasmobackupdb.sqlite3) it shows every single one of my devices has a mac address saved.
Since I have my HA install back up and running again and my iot network setup, I'll see if I can figure out what is going on, but I have been unable to replicate the problem.
